### PR TITLE
chore: improve naming convention and extract common methods

### PR DESCRIPTION
**Pull request by thorseraq**
If an error is converted to `JwstError`, we now use debug format to display it, this will append original error message to current context error message

### CODE REVIEW

File: libs/jwst-storage/src/storage/docs/database.rs [hunk 1]
```diff
@@ -35,14 +36,26 @@ fn migrate_update(updates: Vec<<Docs as EntityTrait>::Model>, doc: Doc) -> JwstR
...
 }
 
+fn merge_doc_records(update_records: Vec<<Docs as EntityTrait>::Model>) -> JwstResult<Vec<u8>> {
+    let state_vector = migrate_update(update_records, Doc::default())
+        .and_then(|doc| {
+            Ok(doc
+                .transact()
+                .encode_state_as_update_v1(&StateVector::default())?)
+        })
+        .expect("failed to encode update");
```
Comment (darkskygit) [inline libs/jwst-storage/src/storage/docs/database.rs:46]: use func()? sugar

File: libs/jwst-storage/src/storage/docs/database.rs [hunk 2]
```diff
@@ -35,14 +36,26 @@ fn migrate_update(updates: Vec<<Docs as EntityTrait>::Model>, doc: Doc) -> JwstR
     Ok(doc)
 }
 
+fn merge_doc_records(update_records: Vec<<Docs as EntityTrait>::Model>) -> JwstResult<Vec<u8>> {
+    let state_vector = migrate_update(update_records, Doc::default())
+        .and_then(|doc| {
+            Ok(doc
+                .transact()
+                .encode_state_as_update_v1(&StateVector::default())?)
```
Comment (darkskygit) [inline libs/jwst-storage/src/storage/docs/database.rs:44]: same as line 32⬆️

File: libs/jwst-storage/src/storage/docs/database.rs [hunk 3]
```diff
@@ -71,142 +84,136 @@ impl DocDBStorage {
...
 
             let data = tokio::task::spawn_blocking(move || {
-                migrate_update(data, Doc::default())
-                    .and_then(|doc| {
-                        Ok(doc
-                            .transact()
-                            .encode_state_as_update_v1(&StateVector::default())?)
-                    })
-                    .expect("failed to encode update")
+                merge_doc_records(doc_records).expect("failed to encode update")
```
Comment (darkskygit) [inline libs/jwst-storage/src/storage/docs/database.rs:181]: consider exposing the result outside the `spawn_blocking`, using func()? syntax sugar process it

File: libs/jwst/src/types.rs [hunk 4]
```diff
@@ -14,9 +14,9 @@ pub enum JwstError {
     BoxedError(#[from] anyhow::Error),
     #[error(transparent)]
     StorageError(anyhow::Error),
-    #[error("io error")]
+    #[error("io error: {0}")]
```
Comment (darkskygit) [inline libs/jwst/src/types.rs:17]: {0} not need in Err(#[from] xxx) define

File: libs/jwst/src/types.rs [hunk 5]
```diff
@@ -14,9 +14,9 @@ pub enum JwstError {
...
     Io(#[from] std::io::Error),
-    #[error("doc codec error")]
+    #[error("doc codec error: {0}")]
```
Comment (darkskygit) [inline libs/jwst/src/types.rs:19]: {0} not need in Err(#[from] xxx) define

**Review by darkskygit — CHANGES_REQUESTED** [review]

**Comment by thorseraq** [comment]
updated at https://github.com/toeverything/OctoBase/pull/347/commits/7c49ba54b1bbc1f79578d60894a89154dada16ab

/cc @darkskygit

**Review by darkskygit — APPROVED** [review]

**Pull request merged**